### PR TITLE
Add function to compare snapshot elements

### PR DESCRIPTION
**Pull request by mint-thompson**
Completes task [CIMPL-1013](https://standardhealthrecord.atlassian.net/browse/CIMPL-1013).

This function takes two StructureDefinitions and returns a list containing element id pairings. Currently, elements are only paired if they have the same id. Eventually, it may be correct to pair extension elements with different ids if their extension URLs are the same.

The function isn't attached to any review processes yet. It just exists nicely and passes its test cases.

### CODE REVIEW

File: test/utils/Comparison.test.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import fs from 'fs-extra';
+import path from 'path';
```
Comment (cmoesel) [inline test/utils/Comparison.test.ts:2]: I noticed that when I run the tests, I see error logs fly across the console.  Usually we try to suppress these for tests by doing something like:
```ts
import '../testhelpers/loggerSpy';
```

Or, if appropriate, actually pull the spy from the import and use it to verify those error logs in the tests that produce them.

**Review by cmoesel — CHANGES_REQUESTED** [review]
Thanks for this, @mint-thompson!

I identified one small thing in the test code that I think should be addressed before merging this.

I also noticed that this doesn't take into account elements that _would_ exist in A or B if we unfolded element types. E.g., if A has `Patient.identifier.system` because it marks it MS, but B only has `Patient.identifier`, then the pairing will not have `Patient.identifier.system` on the B-side. In practice, though, we'd take B's `Patient.identifier` and "unfold" it to get all the `Patient.identifier` sub-elements so we could compare them.  I don't think you need to fix this in this PR though, as I was already planning on implementing an "unfold" type operation in mine; so once I do that, it should be fairly easy to update this code to use it.  I'm only mentioning it here so we're all on the same page about how this function will _eventually_ work.

File: test/utils/Comparison.test.ts (hunk 1 above)
Comment (mint-thompson) [reply test/utils/Comparison.test.ts:2]: @cmoesel Thank you for the suggestion. I've added assertions to the tests that check the error log in the cases where no comparison is possible.

**Review by jafeltra — APPROVED** [review]
Makes sense to me!

**Pull request merged**